### ISSUE TITLE
rm() file after CSV.File fails on Windows 10 but works on MacOS

### ISSUE THREAD

**Issue by orhanabar** [JuliaData/CSV.jl#383]
```
julia> df = CSV.File("test.csv") |> DataFrame
3×2 DataFrame
│ Row │ a      │ b      │
│     │ Int64⍰ │ Int64⍰ │
├─────┼────────┼────────┤
│ 1   │ 1      │ 2      │
│ 2   │ 3      │ 4      │
│ 3   │ 5      │ 6      │

julia> rm("test.csv")
ERROR: IOError: unlink: resource busy or locked (EBUSY)
Stacktrace:
 [1] uv_error at .\libuv.jl:85 [inlined]
 [2] unlink(::String) at .\file.jl:727
 [3] #rm#9(::Bool, ::Bool, ::Function, ::String) at .\file.jl:253
 [4] rm(::String) at .\file.jl:245
 [5] top-level scope at none:0
```
When I run these 2 lines on mac, it does not give any error. However, Windows 10 gives this error. 
I am not sure if I should have created an issue on Julia page. Can you check if you get the same error?

**Comment by nalimilan** [comment]
Windows locks open files, so you can't remove them. Calling `GC.gc()` should hopefully close the file handle if there's no reference to it.

@quinnj Could `CSV.File` support `close`, and `CSV.read` close the object when done?

**Comment by orhanabar** [comment]
After I read the file and store it in dataframe, file should not stay open right?

**Comment by nalimilan** [comment]
The problem is that the `CSV.File` object has no way to know that you're not going to reuse it. If you saved it to a variable, you could collect it to another data frame several times. Only when the GC decides to destroy the object can the file be closed. That's why using `CSV.read` could in theory avoid this problem.

I wonder whether `CSV.File` could automatically close the file when the last line has been reached in `iterate`. It could automatically reopen it if `iterate` is called from the beginning again.

**Comment by quinnj** [comment]
This should be fixed after #395 as we make our own mmapped copy of the file
